pkg/kubelet/status/status_manager.go

```diff
@@ -986,7 +994,7 @@ func mergePodStatus(oldPodStatus, newPodStatus v1.PodStatus, couldHaveRunningCon
 	// while resources that have participanting components above the API use the pod's
 	// transition to a terminal phase (or full deletion) to release those resources.
 	if transitioningToTerminalPhase {
-		if couldHaveRunningContainers {
+		if couldHaveRunningContainers || mightNeedToUnprepareResources {
```

Comment:
Based on comment above we should just change this to:
```
if delayTransitioningToTerminal {
```
Also, don't forget to update the comment to accommodate this change.

Reply:
Thank you for the review! done.